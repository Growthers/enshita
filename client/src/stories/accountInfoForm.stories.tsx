// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
