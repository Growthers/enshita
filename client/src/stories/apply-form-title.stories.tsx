import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
