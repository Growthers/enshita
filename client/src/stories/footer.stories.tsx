// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
