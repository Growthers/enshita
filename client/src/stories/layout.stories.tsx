import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
