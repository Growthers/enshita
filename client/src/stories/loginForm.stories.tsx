// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
