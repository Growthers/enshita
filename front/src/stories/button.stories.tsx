import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
