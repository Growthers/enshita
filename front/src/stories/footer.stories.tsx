import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
