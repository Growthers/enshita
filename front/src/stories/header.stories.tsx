import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Header from "~/components/header";

type T = typeof Header;
type Story = ComponentStory<T>;

export default {
  title: "Header",

  component: Header,
} as ComponentMeta<T>;

export const Default: Story = {};
