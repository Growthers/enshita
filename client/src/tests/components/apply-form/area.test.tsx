import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/apply-form/area.stories";

const { Default, Valid } = composeStories(stories);

describe("(components) text area input with using react-hook-form", () => {
  test("label should be paragraph", () => {
    const { getByLabelText } = render(<Default />);
    expect(getByLabelText("paragraph")).toBeInTheDocument();
  });
  test("(valid) input is correctly entered", async () => {
    const { container, getByLabelText } = render(<Valid />);
    await Valid.play({ canvasElement: container });
    const input = getByLabelText("paragraph") as HTMLInputElement;
    expect(input.value).toEqual("abcdefghij");
  });
  test("take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
