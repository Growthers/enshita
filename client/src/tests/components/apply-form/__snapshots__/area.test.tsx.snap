// Vitest Snapshot v1

exports[`(components) text area input with using react-hook-form > take snapshot 1`] = `
<div>
  <form>
    <section
      class="_enshita-apply-form-area-box_1gqwg_1"
    >
      <label
        class="_enshita-apply-form-area-label_1gqwg_18"
        for="paragraph"
      >
        自由記載欄
      </label>
      <textarea
        aria-invalid="false"
        aria-label="paragraph"
        class="_enshita-apply-form-area_1gqwg_1"
        id="paragraph"
        name="paragraph"
      />
    </section>
  </form>
</div>
`;
